docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/AnomalyDetection/RandomizedPcaSample.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.AnomalyDetection
+{
+    public static class RandomizedPcaSample
+    {
+        /// <summary>
+        /// Example with 3 feature values.
+        /// </summary>
+        private class DataPoint
+        {
+            [VectorType(3)]
+            public float[] Features;
+        }
+
+        /// <summary>
+        /// Class used to capture prediction of <see cref="DataPoint"/> in <see cref="Example"/>.
+        /// </summary>
+        // We disable this warning because complier doesn't realize those fields below are assigned somewhere.
+#pragma warning disable 649
+        private class Result
+        {
+            // Outlier gets false while inlier has true.
+            public bool PredictedLabel;
+            // Outlier gets smaller score.
+            public float Score;
+        }
+#pragma warning restore 649
+
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            var samples = new List<DataPoint>()
+            {
+                new DataPoint(){ Features= new float[3] {1, 0, 0} },
+                new DataPoint(){ Features= new float[3] {0, 2, 1} },
+                new DataPoint(){ Features= new float[3] {1, 2, 3} },
+                new DataPoint(){ Features= new float[3] {0, 1, 0} },
+                new DataPoint(){ Features= new float[3] {0, 2, 1} },
+                new DataPoint(){ Features= new float[3] {-100, 50, -100} }
+            };
+
+            // Convert native C# class to IDataView, a consumble format to ML.NET functions.
```

Comment:
>native C# class [](start = 23, length = 15)

the List<DataPoint>

Reply:
Supper good to make it strongly-typed. No ambiguity anymore.

---
In reply to: [261047430](https://github.com/dotnet/machinelearning/pull/2780#discussion_r261047430) [](ancestors = 261047430)